### PR TITLE
Add analytics

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1233 

## What changed (plus any additional context for devs)
Adds analytical events to track critical swaps path

### CODE REVIEW

File: src/__swaps__/screens/Swap/components/GasButton.tsx [hunk 1]
```diff
@@ -87,26 +90,33 @@ const GasMenu = ({ backToReview = false, children }: { backToReview?: boolean; c
...
 
   const handlePressMenuItem = useCallback(
-    ({ nativeEvent: { actionKey } }: any) => handlePressSpeedOption(actionKey),
+    ({ nativeEvent: { actionKey } }: OnPressMenuItemEventObject) => handlePressSpeedOption(actionKey as GasSpeed),
     [handlePressSpeedOption]
   );
 
   const handlePressActionSheet = useCallback(
-    (buttonIndex: number) => handlePressSpeedOption(menuOptions[buttonIndex]),
+    (buttonIndex: number) => handlePressSpeedOption(menuOptions[buttonIndex] as GasSpeed),
```
Comment (greg-schrammel) [inline src/__swaps__/screens/Swap/components/GasButton.tsx:119]: I don't like this casting `as GasSpeed` ideally it shouldn't be necessary

**Review by greg-schrammel — APPROVED** [review]

File: src/__swaps__/screens/Swap/components/FlipButton.tsx [hunk 2]
```diff
@@ -12,20 +12,27 @@ import { AnimatedBlurView } from '@/__swaps__/screens/Swap/components/AnimatedBl
...
+      analyticsV2.track(analyticsV2.event.swapsFlippedAssets, {
+        inputAmount: SwapInputController.inputValues.value.inputAmount,
+        previousInputAsset: internalSelectedInputAsset.value,
+        previousOutputAsset: internalSelectedOutputAsset.value,
+      });
+
       setAsset({ type: assetTypeToSet, asset: assetToSet });
     }
-  }, [internalSelectedInputAsset, internalSelectedOutputAsset, /* lastTypedInput, */ setAsset]);
+  }, [SwapInputController.inputValues.value, internalSelectedInputAsset.value, internalSelectedOutputAsset.value, setAsset]);
```
Comment (christianbaroni) [inline src/__swaps__/screens/Swap/components/FlipButton.tsx:35]: These dependencies should not include `.value`

File: src/__swaps__/screens/Swap/components/GasButton.tsx [hunk 3]
```diff
@@ -8,19 +9,21 @@ import ContextMenuButton from '@/components/native-context-menu/contextMenu';
 import { Box, Inline, Text, TextIcon, useColorMode, useForegroundColor } from '@/design-system';
 import { IS_ANDROID } from '@/env';
 import * as i18n from '@/languages';
-import { useSwapsStore } from '@/state/swaps/swapsStore';
+import { swapsStore, useSwapsStore } from '@/state/swaps/swapsStore';
```
Comment (christianbaroni) [inline src/__swaps__/screens/Swap/components/GasButton.tsx:12]: Let's pick one

File: src/__swaps__/screens/Swap/components/GasButton.tsx [hunk 4]
```diff
@@ -87,26 +90,36 @@ const GasMenu = ({ backToReview = false, children }: { backToReview?: boolean; c
...
   const customGasSettings = useCustomGasSettings(chainId);
 
-  const menuOptions = useMemo(() => [...keys(metereologySuggestions.data), 'custom'] as const, [metereologySuggestions.data]);
+  const menuOptions = useMemo(() => [...keys(metereologySuggestions.data), GasSpeed.CUSTOM] as GasSpeed[], [metereologySuggestions.data]);
 
   const handlePressSpeedOption = useCallback(
     (selectedGasSpeed: GasSpeed) => {
-      setSelectedGasSpeed(chainId, selectedGasSpeed);
+      const { inputAsset, outputAsset, quote } = swapsStore.getState();
+      analyticsV2.track(analyticsV2.event.swapsChangedSelectedGasSpeed, {
```
Comment (christianbaroni) [inline src/__swaps__/screens/Swap/components/GasButton.tsx:98]: I think we can just track this when a swap is submitted (the selected speed, or custom, on submit)

File: src/__swaps__/screens/Swap/components/SearchInput.tsx [hunk 5]
```diff
@@ -80,13 +81,21 @@ export const SearchInput = ({
   const onInputSearchQueryChange = useDebouncedCallback(
     (text: string) => {
       userAssetsStore.getState().setSearchQuery(text);
+      analyticsV2.track(analyticsV2.event.swapsSearchedForToken, {
```
Comment (christianbaroni) [inline src/__swaps__/screens/Swap/components/SearchInput.tsx:84]: This is going to be very noisy, seems a little excessive. Would suggest we capture the search query (if there is one) when an asset is selected or on input blur. Same for output search tracking below.

File: src/__swaps__/screens/Swap/hooks/useSwapSettings.ts [hunk 6]
```diff
@@ -22,6 +24,22 @@ export const useSwapSettings = ({ inputAsset }: { inputAsset: SharedValue<Extend
     runOnJS(setFlashbots)(!current);
   };
 
+  const handleTrackAndUpdateSlippage = useCallback(
+    (value: string) => {
+      const { inputAsset, outputAsset, quote } = swapsStore.getState();
+
+      setSlippage(value);
+
+      analyticsV2.track(analyticsV2.event.swapsChangedMaximumSlippage, {
```
Comment (christianbaroni) [inline src/__swaps__/screens/Swap/hooks/useSwapSettings.ts:33]: Can track on swap submit

File: src/__swaps__/screens/Swap/providers/swap-provider.tsx [hunk 7]
```diff
@@ -232,15 +249,25 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
...
 
-    // TODO: Analytics
+    analyticsV2.track(analyticsV2.event.swapsSubmitted, {
+      createdAt: Date.now(),
+      type,
+      parameters,
+      selectedGas,
+      slippage,
+      bridge: isBridge,
+      // TODO: Does this work from the JS thread?
```
Comment (christianbaroni) [inline src/__swaps__/screens/Swap/providers/swap-provider.tsx:259]: Yes, it does. In general it's likely more efficient to read from the shared values than from the store. You can always read a shared value from either thread, changes in the value just don't trigger re-renders. On the JS thread they behave just like refs.

File: src/__swaps__/screens/Swap/providers/swap-provider.tsx [hunk 8]
```diff
@@ -400,6 +427,12 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
           ? internalSelectedInputAsset.value?.uniqueId !== extendedAsset?.uniqueId
           : internalSelectedOutputAsset.value?.uniqueId !== extendedAsset?.uniqueId;
 
+      analyticsV2.track(analyticsV2.event.swapsSelectedAsset, {
```
Comment (christianbaroni) [inline src/__swaps__/screens/Swap/providers/swap-provider.tsx:430]: Probably a minor difference but I'd put this at the end of the setAsset function

File: src/__swaps__/screens/Swap/components/SearchInput.tsx (hunk 5 above)
Comment (brunobar79) [reply src/__swaps__/screens/Swap/components/SearchInput.tsx:84]: Yeah, I agree. Even if it's just adding to the batch (and not making a network request) it's not something we want to track on every key press for performance reasons and data pollution on the analytics side.

**Review by brunobar79 — COMMENTED** [review]
Looks good!  Following up on Christian's comments, the takeway is there's certain events where we only want to know the (final) values right at the time execution. 

That's enough data to derive stuff like slippage, where we could create a query on the analytics platform to see if people are using a different value than the default, etc.

Same for the search query, the final result is what matter.

**Review by brunobar79 — APPROVED** [review]
LGTM!

**Review by benisgold — APPROVED** [review]

**Pull request merged**